### PR TITLE
Milestone: preserve data in the browser

### PR DESCRIPTION
**Pull request by jicamargo**
# Milestone: preserve data in the browser

- When the user changes the any input field, the data is saved to the local storage.
- When loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

👍 Thank for any suggestion to improve this project

### CODE REVIEW

File: js/storage-info.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  form.elements.full_name.addEventListener('change', saveInfo);
+  form.elements.email.addEventListener('change', saveInfo);
+  form.elements.message.addEventListener('change', saveInfo);
+}
+
+/** ***** get user info when page loads ****** */
+getInfo();
+
+/** ***** get user info when page loads ****** */
+setEvents();
```
Comment (ABDULALI3468) [inline js/storage-info.js]: ✅ You did literally a great job but the problem is the local storage updates only when the user focuses out of the input element. How it should be done whenever there is a change happens to the input element. Currently, it updates the local state when either it focuses out of the input element or when the user submits the form button so please consider making this change and using the on-change event of JavaScript to do so and to meet the projector requirements.

**Review by ABDULALI3468 — CHANGES_REQUESTED** [review]
# Changes Required 👍 

Hi @jicamargo @jlcarrascof , 



Indeed you did an exceptional but still, there is room for changes to complete this project. 💯


## Highlights 
☑️ Persisted data in local storage.
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.


 ## Required Changes ♻️

> **_Check the comments below to have a look at the required changes._**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/storage-info.js (hunk 1 above)
Comment (jicamargo) [reply js/storage-info.js]: thank you @ABDULALI3468 👍 
I did the change you suggest. 

>With all due respect I tell you: I think that the requirement was achieved, as I developed it, I don't think it is necessary to store the information every time a key is typed, it is better when the final value of the element has changed. (the end result is the same and less resource is consumed).
> Besides, you told me to consider the "on-change" event, and that was what i actually used. For the requirement you ask, i must use "input"

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🥳

Hi @team 👋

Great work implementing the changes suggested by the reviewer. I would like to express my 
deep appreciation for the time and effort you dedicated to this project 💪

**_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Persisted data in local storage., perfect!
-  ✅ Great work using the right GitHub Flow 
-  ✅ Your README looks professional, well done!
-  ✅ Commits are descriptive, good work.
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jlcarrascof** [comment]
Congrats @jicamargo enjoy your meeting tomorrow.

Javier.-

**Pull request merged**